inc/commondbtm.class.php

```diff
@@ -271,20 +271,43 @@ public static function getFromIter(DBmysqlIterator $iter) {
     * @return boolean|array
     */
    public function getFromDBByCrit(array $crit) {
+      return $this->getFromDBByRequest(['WHERE' => $crit]);
+   }
+
+   /**
+    * Retrieve an item from the database by request. The request is an array
+    * similar to the one expected in DB::request().
+    *
+    * @since 9.3
+    *
+    * @see DB::request()
+    *
+    * @param array $request expression
+    *
+    * @return boolean true if succeed else false
+    **/
+   public function getFromDBByRequest(array $request) {
       global $DB;
 
-      $crit = ['SELECT' => 'id',
-               'FROM'   => $this->getTable(),
-               'WHERE'  => $crit];
+      // Limit the request to the useful expressions
+      $request = array_intersect_key($request, [
+         'WHERE' => '',
```

Comment:
I guess we should find a better way to get possible keys here; from iterator itself maybe; to avoid missing entries (order is already missing at least; but if we add new keys in the future, they'll be missing here :/ Any opinions?

Reply:
I considered that order would b e useless. However, reading your comment, this remindss me I did not wanted to allow START and LIMIT, and finally thought this may be useful. Then, ORDER may be useful too.

This array is actually a whitelist. If we do the reverse (a blacklist) we may solve your point.